### PR TITLE
Implement basic API steps and logic

### PR DESCRIPTION
**Pull request by KirillBogdanets**

### CODE REVIEW

File: src/apiVerificationSteps.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * @param {String} responseElement json path
+ * @param {String} action should be named as expected action (equal to|less than|greater)
+ * @param {String} expectedValue Number for comparing with array size
+ */
+Then('Response {text} size is {text} {text}', (pathQuery, action, expectedValue) => {
+  const response = memory.getValue(`${browser.sessionId}response`);
+  const property = jp.value(response.body, pathQuery);
+  const count = property.length;
+
+  switch (action) {
```
Comment (AlexGalichenko) [inline src/apiVerificationSteps.ts:86]: validation is already a part of config loader as parameter type {validation}. So definition may be changed to
Response {text} size {validation} {text}

File: src/apiVerificationSteps.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import memory from '@qavajs/memory';
+import { Then } from '@cucumber/cucumber';
+import jp from 'jsonpath';
```
Comment (AlexGalichenko) [inline src/apiVerificationSteps.ts:3]: will discuss regarding jsonpath. Memory supports traversal througth object OOB

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Before } from '@cucumber/cucumber';
```
Comment (AlexGalichenko) [inline src/index.ts:1]: we can remove this file. Config loads in config-loader before hook.

File: src/apiActionSteps.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+ * @param url should be API endpoint
+ */
+When('I send {text} request to {landing-url}', async function (method: string, url: string) {
+  const conf: RequestInit = {
+    method,
+  };
+  const response = await sendHttpRequest(url, conf);
+
+  // store response to memory to be able to use it in next steps
+  memory.setValue(`${browser.sessionId}response`, response);
```
Comment (AlexGalichenko) [inline src/apiActionSteps.ts:22]: maybe we need to give user ability to save responce with custom name. And there maybe a case when browser is not available (in pure API) project

File: src/parameterTypes.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { defineParameterType } from '@cucumber/cucumber';
+import { getTestDataFilePath } from '@utils';
+import fse from 'fs-extra';
```
Comment (AlexGalichenko) [inline src/parameterTypes.ts:3]: Will discuss if it possible to use built-in constants instead of file reading

Comment (KirillBogdanets) [reply src/parameterTypes.ts:3]: already discussed

File: index.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+/* eslint-disable import/no-unresolved */
```
Comment (AlexGalichenko) [inline index.js:1]: need to list exported stuff here. As I understand it should be actionsSteps, verificationSteps, hooks?, parameterTypes

File: package.json [hunk 7]
```diff
@@ -57,8 +57,6 @@
     "@qavajs/memory": "^1.0.0",
     "chai": "^4.3.6",
     "fs-extra": "^10.1.0",
-    "jsonpath": "^1.1.1",
-    "node-fetch": "^3.2.6",
-    "webdriverio": "^7.20.5"
+    "node-fetch": "^3.2.6"
```
Comment (AlexGalichenko) [inline package.json:60]: pls move @qavajs/memory and node-fetch to dependencies

Comment (AlexGalichenko) [reply package.json:60]: chai too

File: index.js (hunk 6 above)
Comment (KirillBogdanets) [reply index.js:1]: done

File: package.json (hunk 7 above)
Comment (KirillBogdanets) [reply package.json:60]: discussed

**Pull request merged**